shared/chain_support.go

```diff
@@ -1,4 +1,4 @@
-package commit
+package shared
```

Comment:
Moving it to shared so that we can use without cyclic dependencies. Just changing fields to make them publicly available. No added or updated functionality.